tools/generators/normalize-package-dependencies/index.spec.ts

```diff
@@ -0,0 +1,314 @@
+import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
+import {
+  Tree,
+  addProjectConfiguration,
+  writeJson,
+  getProjects,
+  readJson,
+  joinPathFragments,
+  ProjectGraph,
+  readProjectConfiguration,
+  updateJson,
+} from '@nrwl/devkit';
+import * as chalk from 'chalk';
+
+import generator from './index';
+import { PackageJson } from '../../types';
+import { disableChalk } from '../../utils-testing';
+
+const graphMock: ProjectGraph = {
+  dependencies: {},
+  nodes: {},
+  externalNodes: {},
+};
+
+jest.mock('@nrwl/devkit', () => {
+  async function createProjectGraphAsyncMock(): Promise<ProjectGraph> {
+    return graphMock;
+  }
+
+  return {
+    ...jest.requireActual('@nrwl/devkit'),
+    createProjectGraphAsync: createProjectGraphAsyncMock,
```

Comment:
need to mock this one as it doesnt work with `tree`, thus getting results from actual workspace. this is by design -> needs to be mocked in test.